File: hw/codena/u1/hw06Files/Python/model/file_manager.py
import csv
import json
import os

from model.student import Student
from model.teacher import Teacher


# SAVE STUDENT CSV
def save_student_csv(student):

    with open("students.csv", "a", newline="") as file:

        writer = csv.writer(file)

        writer.writerow([
            student.id,
            student.name,
            student.course
        ])


# SAVE TEACHER CSV
def save_teacher_csv(teacher):

    with open("teachers.csv", "a", newline="") as file:

        writer = csv.writer(file)

        writer.writerow([
            teacher.id,
            teacher.name,
            teacher.career
        ])


# READ STUDENTS CSV
def read_students_csv():

    students = []

    if not os.path.exists("students.csv"):
        return students

    with open("students.csv", "r") as file:

        reader = csv.reader(file)

        for row in reader:

            student = Student(
                int(row[0]),
                row[1],
                row[2]
            )

            students.append(student)

    return students


# READ TEACHERS CSV
def read_teachers_csv():

    teachers = []

    if not os.path.exists("teachers.csv"):
        return teachers

    with open("teachers.csv", "r") as file:

        reader = csv.reader(file)

        for row in reader:

            teacher = Teacher(
                int(row[0]),
                row[1],
                row[2]
            )

            teachers.append(teacher)

    return teachers


# SAVE STUDENT JSON
def save_student_json(student):

    students_data = []

    if os.path.exists("students.json"):

        with open("students.json", "r") as file:

            try:
                students_data = json.load(file)

            except json.JSONDecodeError:
                students_data = []

    students_data.append(student.to_dict())

    with open("students.json", "w") as file:

        json.dump(students_data, file, indent=4)


# SAVE TEACHER JSON
def save_teacher_json(teacher):

    teachers_data = []

    if os.path.exists("teachers.json"):

        with open("teachers.json", "r") as file:

            try:
                teachers_data = json.load(file)

            except json.JSONDecodeError:
                teachers_data = []

    teachers_data.append(teacher.to_dict())

    with open("teachers.json", "w") as file:

        json.dump(teachers_data, file, indent=4)


# READ STUDENTS JSON
def read_students_json():

    students = []

    if not os.path.exists("students.json"):
        return students

    with open("students.json", "r") as file:

        data = json.load(file)

        for item in data:

            student = Student(
                item["id"],
                item["name"],
                item["course"]
            )

            students.append(student)

    return students

# READ TEACHERS JSON
def read_teachers_json():

    teachers = []

    if not os.path.exists("teachers.json"):
        return teachers

    with open("teachers.json", "r") as file:

        data = json.load(file)

        for item in data:

            teacher = Teacher(
                item["id"],
                item["name"],
                item["career"]
            )

            teachers.append(teacher)

    return teachers

File: hw/codena/u1/hw06Files/JavaScript/model/FileManager.js
import fs from "fs";

import Student from "./Student.js";
import Teacher from "./Teacher.js";

// SAVE STUDENT CSV
export function saveStudentCSV(student) {

    const data =
        `${student.id},${student.name},${student.course}\n`;

    fs.appendFileSync("students.csv", data);
}

// SAVE TEACHER CSV
export function saveTeacherCSV(teacher) {

    const data =
        `${teacher.id},${teacher.name},${teacher.career}\n`;

    fs.appendFileSync("teachers.csv", data);
}

// READ STUDENTS CSV
export function readStudentsCSV() {

    let students = [];

    if (!fs.existsSync("students.csv")) {
        return students;
    }

    const data =
        fs.readFileSync("students.csv", "utf-8");

    const lines = data.split("\n");

    for (let line of lines) {

        if (line.trim() !== "") {

            const values = line.split(",");

            const student = new Student(
                Number(values[0]),
                values[1],
                values[2]
            );

            students.push(student);
        }
    }

    return students;
}

// READ TEACHERS CSV
export function readTeachersCSV() {

    let teachers = [];

    if (!fs.existsSync("teachers.csv")) {
        return teachers;
    }

    const data =
        fs.readFileSync("teachers.csv", "utf-8");

    const lines = data.split("\n");

    for (let line of lines) {

        if (line.trim() !== "") {

            const values = line.split(",");

            const teacher = new Teacher(
                Number(values[0]),
                values[1],
                values[2]
            );

            teachers.push(teacher);
        }
    }

    return teachers;
}

// SAVE STUDENT JSON
export function saveStudentJSON(student) {

    let students = [];

    if (fs.existsSync("students.json")) {

        const data =
            fs.readFileSync("students.json", "utf-8");

        if (data.trim() !== "") {
            students = JSON.parse(data);
        }
    }

    students.push(student);

    fs.writeFileSync(
        "students.json",
        JSON.stringify(students, null, 4)
    );
}

// SAVE TEACHER JSON
export function saveTeacherJSON(teacher) {

    let teachers = [];

    if (fs.existsSync("teachers.json")) {

        const data =
            fs.readFileSync("teachers.json", "utf-8");

        if (data.trim() !== "") {
            teachers = JSON.parse(data);
        }
    }

    teachers.push(teacher);

    fs.writeFileSync(
        "teachers.json",
        JSON.stringify(teachers, null, 4)
    );
}

// READ STUDENTS JSON
export function readStudentsJSON() {

    let students = [];

    if (!fs.existsSync("students.json")) {
        return students;
    }

    const data =
        fs.readFileSync("students.json", "utf-8");

    if (data.trim() === "") {
        return students;
    }

    const jsonData = JSON.parse(data);

    for (let item of jsonData) {

        const student = new Student(
            item.id,
            item.name,
            item.course
        );

        students.push(student);
    }

    return students;
}

// READ TEACHERS JSON
export function readTeachersJSON() {

    let teachers = [];

    if (!fs.existsSync("teachers.json")) {
        return teachers;
    }

    const data =
        fs.readFileSync("teachers.json", "utf-8");

    if (data.trim() === "") {
        return teachers;
    }

    const jsonData = JSON.parse(data);

    for (let item of jsonData) {

        const teacher = new Teacher(
            item.id,
            item.name,
            item.career
        );

        teachers.push(teacher);
    }

    return teachers;
}